### DISCUSSION TITLE
Add to collection without initialization

### DISCUSSION THREAD

**Discussion by melonges** [mikro-orm/mikro-orm#5864] [Q&A] [answered]
i want to add new record, but why  I need to download all the records from the database to take advantage of this nice method?

```ts
   const newReferral = new Referral(referrer, referee);
   await referrer.referrals.load() // ?
   referrer.referrals.add(newReferral);
```

**Comment by B4nan** [comment]
you do not have to init the collection to be able to add items to it

**Reply by melonges** [reply]
in docs: _Note that array access in Collection is available only for reading already loaded items, we cannot add new items to Collection this way._

and I got error if the collection is empty `Error in middleware while handling update 346462304 Error: Collection<Asset> of entity Player[1202446984] not initialized
`

**Reply by B4nan** [reply, marked as answer]
> Note that array access in Collection is available only for reading already loaded items, we cannot add new items to Collection this way.

You are reading too much in between the lines. It says, you can't add items to collections via array access, that has nothing to do with the `add` method, it just means you can't assign to the collection like `author.books[0] = new Book()` (but you can access the first item via `author.books[0]`), you need to use `author.books.add(new Book())`.

> and I got error if the collection is empty Error in middleware while handling update 346462304 Error: Collection<Asset> of entity Player[1202446984] not initialized 

This is not caused by adding to a collection.
